certora/specs/AllowancesInvariant.spec

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+using GeneralAdapter1 as GeneralAdapter1;
+using EthereumGeneralAdapter1 as EthereumGeneralAdapter1;
+using ParaswapAdapter as ParaswapAdapter;
+
+methods {
+    function _.approve(address token, address spender, uint256 amount)  external => summaryApprove(calledContract, spender, amount) expect bool;
+    // Aave dispatch
+    function _.repay(address, uint256, uint256, address) external => HAVOC_ECF;
+    // Compound dispatch
+    function _.repayBorrowBehalf(address, uint256) external => HAVOC_ECF;
+    function _.underlying() external => mockUnderlying() expect address;
+    function _.supplyTo(address, address, uint256) external => HAVOC_ECF;
+    function _.baseToken() external => mockBaseToken() expect address;
+    function _.compoundV2RedeemEth(uint256,address) external => HAVOC_ECF;
+    // Paraswap dispatch
+    function _.set(bytes memory, uint256 offset, uint256) internal => setData(offset) expect void;
+    function _.get(bytes memory, uint256 offset) internal => getData(offset) expect uint256;
+    function _.isValidAugustus(address) external => NONDET;
+    // ERC20
+    function _.transfer(address, uint256) external => DISPATCHER(true);
+    function _.balanceOf(address) external => DISPATCHER(true);
+    unresolved external in _.nativeTransfer(address, uint256) => DISPATCH [] default HAVOC_ECF;
+    unresolved external in _.unwrapNative(uint256,address) => DISPATCH [] default HAVOC_ECF;
+    unresolved external in _.compoundV2RedeemEth(uint256, address) => DISPATCH [] default HAVOC_ECF;
+    unresolved external in _.onMorphoRepay(uint256, bytes) => DISPATCH [] default HAVOC_ECF;
+    unresolved external in _.onMorphoSupply(uint256, bytes) => DISPATCH [] default HAVOC_ECF;
+    unresolved external in _.onMorphoSupplyCollateral(uint256, bytes) => DISPATCH [] default HAVOC_ECF;
+    unresolved external in _.onMorphoFlashLoan(uint256, bytes) => DISPATCH [] default HAVOC_ECF;
+    unresolved external in _._ => DISPATCH [] default ASSERT_FALSE;
+}
+
+persistent ghost address lastErc20Underlying;
+persistent ghost bool erc20UnderlyingChanged;
+
+function mockUnderlying() returns address {
+    address erc20;
+    if (erc20 != lastErc20Underlying) {
+        erc20UnderlyingChanged = true;
+        lastErc20Underlying = erc20;
+    }
+    return erc20;
+}
+
+persistent ghost address lastErc20BaseToken;
+persistent ghost bool erc20BaseTokenChanged;
+
+function mockBaseToken() returns address {
+    address erc20;
+    if (erc20 != lastErc20BaseToken) {
+        erc20BaseTokenChanged = true;
+        lastErc20BaseToken = erc20;
+    }
+    return erc20;
+}
```

Comment:
```suggestion
```
we can remove those now, since they are not used in the spec anymore

Reply:
[Run](https://prover.certora.com/output/66603/6b07e841a705465a830fb4e67f1ac3f8?anonymousKey=30d9e27a1307b47e7af9379b676fac689867eb76) without it